### PR TITLE
JAVA-863: [3.0] Idempotence propagation in PreparedStatements.

### PR DESCRIPTION
**Pull request by newkek**

### CODE REVIEW

**Comment by tolbertam** [comment]
Looks good to me other than the 2 small comments I had.  :+1:

**Comment by adutra** [comment]
:+1: 
Please add:
- an entry to the changelog
- an item to the upgrade guide explaining that IdempotenceAwarePreparedStatement has been removed

File: driver-core/src/test/java/com/datastax/driver/core/PreparedStatementTest.java [hunk 1]
```diff
@@ -664,4 +664,42 @@ private void assertEventsContain(QueryTrace queryTrace, String toFind) {
...
+        PreparedStatement prepared;
+        BoundStatement bound;
+
+        statement = session.newSimpleStatement(String.format("SELECT * FROM %s.idempotencetest WHERE i = ?", keyspace));
+
+        prepared = session.prepare(statement);
+        bound = prepared.bind(1);
+
+//      This may not work if the default idempotence for regular statements change.
+//      Which is likely to happen with JAVA-819
```
Comment (olim7t) [inline driver-core/src/test/java/com/datastax/driver/core/PreparedStatementTest.java]: Mmm I'm not sure.

**Comment by olim7t** [comment]
:+1:

File: driver-core/src/test/java/com/datastax/driver/core/PreparedStatementTest.java (hunk 1 above)
Comment (adutra) [inline driver-core/src/test/java/com/datastax/driver/core/PreparedStatementTest.java]: Default idempotence will not change with JAVA-819. Please remove this comment.

Comment (newkek) [inline driver-core/src/test/java/com/datastax/driver/core/PreparedStatementTest.java]: Yes it was just a reminder for me, I'll put that out.

**Comment by newkek** [comment]
Squashed and rebased on current 3.0.

**Pull request merged**